src/transaction/build.rs

```diff
@@ -94,32 +96,38 @@ impl<T: Transport> BuildFuture<T> {
         };
 
         match builder.from {
-            None => BuildFuture::LocallySigned(LocalBuildFuture::new(
+            None => BuildFuture::Local(BuildTransactionRequestForLocalSigningFuture::new(
                 &builder.web3,
                 None,
-                builder.gas_price,
+                gas_price,
                 TransactionRequestOptions(options, None),
             )),
             Some(Account::Local(from, condition)) => {
-                BuildFuture::LocallySigned(LocalBuildFuture::new(
+                BuildFuture::Local(BuildTransactionRequestForLocalSigningFuture::new(
                     &builder.web3,
                     Some(from),
-                    builder.gas_price,
+                    gas_price,
                     TransactionRequestOptions(options, condition),
                 ))
             }
             Some(Account::Locked(from, password, condition)) => {
```

Comment:
Is the password encrypted here? Or is this a silly question?

Reply:
It is not, it’s in protected memory though, which gets set to bunch of 0’s once it’s used to make leaking the value less likely.